TestSwitchApi/TestSwitchApi/Repositories/CandidateTestsRepo.cs

```diff
@@ -25,15 +25,17 @@ public IEnumerable<CandidateTestModel> GetSubmissionsByCandidateId(int id)
 
         public CandidateTestModel AddTestSubmission(int candidateId, AddTestSubmissionRequestModel testSubmissionModel)
         {
-            var insertTest = _context.CandidateTests.Add(new CandidateTestModel()
+            var result = _context.CandidateTests
+                .SingleOrDefault(s => s.CandidateId == candidateId && s.TestId == testSubmissionModel.TestId);
+
+            if (result != null)
```

Comment:
What happens if the result is null?
What _should_ happen? Error? Make a new Entry?

Reply:
The result being null should not occur because we are creating empty tests when a new candidate is created 
but even if it occurs I don't think we should throw an error, instead I have coded it to insert that test